csm/core/services/alerts.py

```diff
@@ -223,6 +224,12 @@ async def retrieve_unresolved_by_node_id(self, node_id) -> Iterable[AlertModel]:
         query = query.limit(limits.limit)
         return await self.db(AlertModel).get(query)
 
+    def set_hostname_in_alert(self, alert):
```

Comment:
You should not pass a complete alert payload for setting up the hostname, it will be an overkill and a costly algorithm. 
The idea is to fetch the hostname from minion id and attach a new key to alerts.
You can have a simple function for getting the hostname from minion id. Pass minion_id only and return the hostname. 
Something like - 
def get_hostname(self, minion_id):
         hostname = map.get(minion_id, "")
         return hostname

Also, do not make changes to AlertsAppService class. It is agnostic. That is why we created a repo class for any specific changes. 
After fetching hostname make changes to repo class to include hostname in alerts and return alerts from repo. Service class should not have any changes.